contracts/truefi2/DebtToken.sol

```diff
@@ -4,37 +4,108 @@ pragma solidity 0.6.10;
-
-    function pool() external override view returns (ITrueFiPool2) {}
-
-    function status() external override view returns (Status) {}
-
-    function redeem(uint256 _amount) external override {}
-
-    function repay(address _sender, uint256 _amount) external override {}
-
-    function repayInFull(address _sender) external override {}
-
-    function reclaim() external override {}
-
-    function liquidate() external override {}
-
-    function repaid() external override view returns (uint256) {}
-
-    function isRepaid() external override view returns (bool) {}
-
-    function balance() external override view returns (uint256) {}
-
-    function value(uint256 _balance) external override view returns (uint256) {}
-
-    function token() external override view returns (ERC20) {}
-
-    function version() external override pure returns (uint8) {}
+    using SafeMath for uint256;
+    using SafeERC20 for ERC20;
+
+    address public override borrower;
+    uint256 public override debt;
+
+    address public liquidator;
+    ITrueFiPool2 public override pool;
+    ERC20 public override token;
+
+    uint256 public redeemed;
+
+    Status public override status;
+
+    /**
+     * @dev Emitted when a DebtToken is redeemed for underlying tokens
+     * @param receiver Receiver of tokens
+     * @param burnedAmount Amount of DebtTokens burned
+     * @param redeemedAmount Amount of token received
+     */
+    event Redeemed(address receiver, uint256 burnedAmount, uint256 redeemedAmount);
+
+    /**
+     * @dev Emitted when loan gets liquidated
+     * @param status Final loan status
+     */
+    event Liquidated(Status status);
+
+    function initialize(
+        ITrueFiPool2 _pool,
+        address _lender,
+        address _borrower,
+        address _liquidator,
+        uint256 _debt
+    ) external initializer {
+        ERC20.__ERC20_initialize("TrueFi Debt Token", "DEBT");
+
+        pool = _pool;
+        token = _pool.token();
+        borrower = _borrower;
+        liquidator = _liquidator;
+        debt = _debt;
+        status = Status.Defaulted;
+        _mint(_lender, _debt);
+    }
+
+    /**
+     * @dev Redeem DebtToken balances for underlying token
+     * Can only call this function after the loan is Closed
+     * @param _amount amount to redeem
+     */
+    function redeem(uint256 _amount) external override {
+        uint256 amountToReturn = _amount.mul(_balance()).div(totalSupply());
+        redeemed = redeemed.add(amountToReturn);
+        _burn(msg.sender, _amount);
+        token.safeTransfer(msg.sender, amountToReturn);
+
+        emit Redeemed(msg.sender, _amount, amountToReturn);
+    }
+
+    /**
+     * @dev Liquidate the loan if it has defaulted
+     */
+    function liquidate() external override {
+        require(status == Status.Defaulted, "DebtToken: Current status should be Defaulted");
+        require(msg.sender == liquidator, "DebtToken: Caller is not the liquidator");
```

Comment:
Nice! I hadn't realized the `onlyDefaulted onlyLiquidator` modifiers were only used for this one function. Better to inline them like this since we're not worried about dedups.